whois-scheduler/src/main/java/net/ripe/db/whois/scheduler/DailySchedulerJmx.java

```diff
@@ -0,0 +1,97 @@
+package net.ripe.db.whois.scheduler;
+
+import com.google.common.collect.Maps;
+import net.ripe.db.whois.common.DateTimeProvider;
+import net.ripe.db.whois.common.jmx.JmxBase;
+import net.ripe.db.whois.common.scheduler.DailyScheduledTask;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.aop.support.AopUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.jmx.export.annotation.ManagedOperation;
+import org.springframework.jmx.export.annotation.ManagedOperationParameter;
+import org.springframework.jmx.export.annotation.ManagedOperationParameters;
+import org.springframework.jmx.export.annotation.ManagedResource;
+import org.springframework.scheduling.TaskScheduler;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.scheduling.support.ScheduledMethodRunnable;
+import org.springframework.stereotype.Component;
+
+import java.lang.reflect.Method;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ScheduledFuture;
+import java.util.stream.Collectors;
+
+@Component
+@ManagedResource(objectName = JmxBase.OBJECT_NAME_BASE + "DailyScheduler", description = "Whois daily scheduler")
+public class DailySchedulerJmx extends JmxBase {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DailySchedulerJmx.class);
+
+    private final TaskScheduler taskScheduler;
+    private Map<String, ScheduledMethodRunnable> scheduledTasks;
+    private final DateTimeProvider dateTimeProvider;
+
+    @Autowired
+    public DailySchedulerJmx(@Qualifier("scheduler") final TaskScheduler taskScheduler,
+                             final DateTimeProvider dateTimeProvider,
+                             final List<DailyScheduledTask> scheduledTasks) {
+        super(LOGGER);
+        this.taskScheduler = taskScheduler;
+        this.dateTimeProvider = dateTimeProvider;
+        this.scheduledTasks = createScheduledMethodRunnables(scheduledTasks);
+    }
+
+    private Map<String, ScheduledMethodRunnable> createScheduledMethodRunnables(final List<DailyScheduledTask> scheduledTasks) {
```

Comment:
Add a comment to explain that you need to wrap all Scheduled classes ?

Reply:
Done